### PR TITLE
MAINT: better cibuildwheel

### PR DESCRIPTION
**Pull request by larsoner**
1. Add dependabot
2. Add `release.yml` to exclude dependabot PRs
3. Update cibuildwheel to latest version
4. Build macos arm64 and intel wheels on native hardware (and test both)

### CODE REVIEW

**Comment by larsoner** [comment]
@akaszynski I don't have the permissions necessary to change the merge requirements / branch protection so you'll have to change them (or give me permissions to do it)

**Comment by akaszynski** [comment]
> I don't have the permissions necessary

Just made you an admin.

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+version: 2
+updates:
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    schedule:
+      interval: "weekly"
```
Comment (akaszynski) [inline .github/dependabot.yml:6]: Would prefer less aggressive update schedule.

```suggestion
      interval: "monthly"
```

**Review by akaszynski — APPROVED** [review]
Corrected the required checks.

Thanks for the PR!

**Comment by larsoner** [comment]
Okay should be good to release then! Feel free if you're up for it @akaszynski or I can do it

**Comment by akaszynski** [comment]
> Okay should be good to release then! Feel free if you're up for it @akaszynski or I can do it

I'll make the release.

**Pull request merged**